SU-HSDO/suhumsci
# STN-458 : Animation Enhancement Toggler

# READY FOR REVIEW

## Summary

Adds a theme setting that allows site owners to toggle site wide animation enhancements. The default state of the Animation Enhancements setting is `false`. Site owners must click a checkbox to enable the theme setting. Once a the Animation Enhancements setting is checked, the class of `hb-has-animation-enhancements` is added to the `<html>` tag.

See [JIRA STN-458](https://sparkbox.atlassian.net/browse/STN-458)

## Need Review By (Date)
8/6/2020

## Urgency
medium

## Steps to Test
1. Go to http://sparkbox-sandbox.suhumsci.loc/
2. Open your browser's developer tools and inspect the classes on the `<html>` tag. You **should not** see the class of `hb-has-animation-enhancements` as the default state of the theme setting is set to `false`.
3. Go to Appearance > click on the `Settings` for the theme you currently have set as the default (Colorful or Traditional --> either one works!)
4. Scroll to the `Theme Specific Settings`, find `Animation Enhancements`, and check the box to `Use animation enhancements`
5. Save and go back to the site's homepage
6. Verify the class of `hb-has-animation-enhancements` has been added to the `<html>` tag.
7. Go back to your theme settings. Change your default theme (Colorful or Traditional). Repeat the steps above.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
